README.md

```diff
@@ -29,6 +30,206 @@ runApp(...
 
 2. Run the application in debug mode and watch for a leak related warnings. If you see a warning, open the link to investigate the leaks.
 
-TODO(polina-c): add example of the warning
+TODO(polina-c): implement the link and add example of the warning.
+
+### Flutter tests
+
+Wrap your tests with `withLeakTracking` to setup automated leak verification:
+
+```dart
+test('...', () async {
+  final leaks = await withLeakTracking(
+    () async {
+      ...
+    },
+    throwOnLeaks: false,
+  );
+
+  expect(leaks.total, 0);
+});
+```
+
+## Leak tracking concepts
+
+Before reading about leak tracking, understand [Dart memory concepts](https://docs.flutter.dev/development/tools/devtools/memory#basic-memory-concepts).
+
+### Addressed leak types
+
+The leak tracker can catch only certain types of leaks, in particular, related to timing of disposal and garbage collection.
+
+The tool assumes that, with proper memory management,
+an object's disposal and garbage collection should happen sequentially,
+close to each other.
+
+By monitoring disposal and GC events, the tool detects different types of leaks:
+
+**Not disposed, but GCed (not-disposed)**: a disposable object was GCed,
+without being disposed first. This means that the object's disposable content
+is using memory after the object is no longer needed.
+To fix the leak, invoke `dispose()` to free up the memory.
+
+**Disposed, but not GCed (not-GCed)**: an object was disposed,
+but not GCed after certain number of GC events. This means that
+a reference to the object is preventing it from being
+garbage collected after it's no longer needed.
+To fix the leak, after disposal assign all reachable references
+of the object to null:
+
+```
+myField.dispose();
+myField = null;
+```
+
+**Disposed and GCed late (GCed-late)**: an object was disposed and then GCed,
+but GC happened later than expected. This means the retaining path was
+holding the object in memory for some period, but then disappeared.
+
+**Disposed, but not GCed, without path (not-GCed-without-path)**: an object
+was disposed and not GCed when expected, but retaining path is not detected,
+that means that the object will be most likely GCed in the next GC cycle,
+and the leak will convert to **GCed-late** leak.
+
+### Culprits and victims
```

Comment:
This terminology may end up being a bit too harsh. It may be worth asking the tech writers if they have different analogies they would prefer using here.

Reply:
Tech writers and UXR already reviewed it and they loved it, because it is extremely illustrative :)